src/storage/src/hummock/compactor/mod.rs

```diff
@@ -593,3 +626,176 @@ pub fn start_compactor(compactor_context: Arc<CompactorContext>) -> (JoinHandle<
+                    });
+                }
+
+                let report_compaction_task_request = ReportCompactionTaskRequest{
+                    event: Some(ReportCompactionTaskEvent::HeartBeat(
+                        SharedHeartBeat {
+                            progress: progress_list
+                        }
+                    )),
+                 };
+
+                //  match client.repory_compaction_task(report_compaction_task_request).await{
+                //     Ok(_) => {},
+                //     Err(e) => tracing::warn!("Failed to report heartbeat"),
+                // }
+            }
+
+           
+            _ = &mut shutdown_rx => {
+                tracing::info!("Compactor is shutting down");
+                return
+            }
+        };
+
+        match dispatch_task {
+            dispatch_compaction_task_request::Task::CompactTask(compact_task) => {
+                running_task_count.fetch_add(1, Ordering::SeqCst);
+                let (tx, rx) = tokio::sync::oneshot::channel();
+                let task_id = compact_task.task_id;
+                shutdown.lock().unwrap().insert(task_id, tx);
+                let mut output_object_ids: VecDeque<_> = VecDeque::new();
+                output_object_ids.extend(output_ids);
+                let sstable_object_id_manager =
+                    SharedComapctorObjectIdManager::new(output_object_ids);
+                let filter_key_extractor_manager =
+                    FilterKeyExtractorManagerFactory::ServerlessFilterKeyExtractorManager(
+                        FilterKeyExtractorBuilder::new(id_to_table),
+                    );
+                let (compact_task, table_stats) = shared_compact(
+                    compactor_metrics,
+                    compact_task,
+                    rx,
+                    sstable_store,
+                    storage_opts,
+                    filter_key_extractor_manager,
+                    worker_num,
+                    memory_limiter,
+                    Box::new(sstable_object_id_manager.clone()),
+                    task_progress_manager,
+                    await_tree_reg,
+                )
+                .await;
+                shutdown.lock().unwrap().remove(&task_id);
+                running_task_count.fetch_sub(1, Ordering::SeqCst);
+                let report_compaction_task_request = ReportCompactionTaskRequest {
+                    event: Some(ReportCompactionTaskEvent::ReportTask(ReportSharedTask {
+                        compact_task: Some(compact_task),
+                        table_stats_change: to_prost_table_stats_map(table_stats),
+                    })),
+                };
+
+                // match client.repory_compaction_task(report_compaction_task_request).await{
+                //     Ok(_) => {},
+                //     Err(e) => tracing::warn!("Failed to report task {task_id:?} . {e:?}"),
+                // }
+            }
```

Comment:
Currently there is no report_compaction_task interface in `HummockManagerService`.